src/frontend/src/icp/components/core/IcLoaderWallets.svelte

```diff
@@ -0,0 +1,71 @@
+<script lang="ts">
+	import { onDestroy } from 'svelte';
+	import { initIcrcWalletWorker } from '$icp/services/worker.icrc-wallet.services';
+	import { initIcpWalletWorker } from '$icp/services/worker.icp-wallet.services';
+	import type { WalletWorker } from '$icp/types/ic-listener';
+	import { enabledIcNetworkTokens } from '$lib/derived/network-tokens.derived';
+	import type { TokenId } from '$lib/types/token';
+	import { debounce, nonNullish } from '@dfinity/utils';
+	import { toastsError } from '$lib/stores/toasts.store';
+	import { i18n } from '$lib/stores/i18n.store';
+	import { replacePlaceholders } from '$lib/utils/i18n.utils';
+
+	let workers: Map<TokenId, WalletWorker> = new Map<TokenId, WalletWorker>();
+
+	const manageWorkers = async () => {
+		const unusedWorkers: TokenId[] = Array.from(workers.keys()).filter(
+			(tokenId) => !$enabledIcNetworkTokens.some(({ id: icTokenId }) => icTokenId === tokenId)
+		);
+
+		unusedWorkers.forEach((tokenId) => {
+			const worker = workers.get(tokenId);
+
+			if (nonNullish(worker)) {
+				worker.stop();
+				workers.delete(tokenId);
+			}
+		});
+
+		await Promise.all(
+			$enabledIcNetworkTokens.map(async (token) => {
+				if (!workers.has(token.id)) {
+					try {
+						const worker = await (token.standard === 'icrc'
+							? initIcrcWalletWorker(token)
+							: initIcpWalletWorker());
+
+						worker.stop();
+						worker.start();
+
+						workers.set(token.id, worker);
+					} catch (err: unknown) {
+						// We show the error to the user, but we keep loading the rest of the workers
+						toastsError({
+							msg: {
+								text: replacePlaceholders($i18n.tokens.error.loading_ic_wallet, {
+									$symbol: token.symbol
+								})
+							},
+							err
+						});
+					}
+				}
+			})
+		);
+	};
+
+	const debounceManageWorkers = debounce(manageWorkers, 500);
+
+	$: $enabledIcNetworkTokens, debounceManageWorkers();
+
+	onDestroy(() => {
+		workers.forEach((worker) => worker.stop());
+		workers.clear();
+	});
+
+	const triggerTimer = () => workers.forEach((worker) => worker.trigger());
```

Comment:
Only the selected worker should be triggered no?

Reply:
Up until now the event `oisyTriggerWallet` is emitted in a few cases but without any specific token associated.

In case it is emitted and the workers is mounted by `IcListener` (meaning that we are in the token list page), it would have triggered all of the workers, because all the `IcListener` are mounted in that page and each one build a different worker.

The same behavior will happen after this PR: the workers of `IcLoaderWallets` are all mounted and dismounted by the token page.

I didn't want to change any behavior or logic in this PR, but I will revisit the necessity of having the manual trigger in the token list page.

NOTE: I am not affecting the workers that are in the `transaction` page.

Reply:
Agree. Maybe something we can improve later.